core/src/replay_stage.rs

```diff
@@ -1154,6 +1154,32 @@ impl ReplayStage {
         })
     }
 
+    fn reload_tower_with_new_identity(
+        tower_storage: &Arc<dyn TowerStorage>,
+        new_identity: &Pubkey,
+        vote_account: &Pubkey,
+        bank_forks: &Arc<RwLock<BankForks>>,
+    ) -> Tower {
+        Tower::restore(tower_storage.as_ref(), new_identity)
+            .and_then(|restored_tower| {
+                let root_bank = bank_forks.read().unwrap().root_bank();
+                let slot_history = root_bank.get_slot_history();
+                restored_tower.adjust_lockouts_after_replay(root_bank.slot(), &slot_history)
+            })
+            .unwrap_or_else(|err| {
+                if err.is_file_missing() {
+                    Tower::new_from_bankforks(
+                        &bank_forks.read().unwrap(),
+                        new_identity,
+                        vote_account,
+                    )
+                } else {
+                    error!("Failed to load tower for {}: {}", new_identity, err);
+                    std::process::exit(1);
```

Comment:
Why exit instead of panic! ? Would returning an error be better?

Reply:
I think it's just rust style guidelines:
"For planned shutdowns, use exit(). Do note that a known error is considered a planned shutdown
For unplanned shutdowns (i.e. exceptional failures) consider panic!(), as you'll both benefit from being able to get a stack trace when this happens, and the failure case should be exceptional enough that it is effectively unaccounted for and stems from an unplanned scenario"
not being able to load tower due to corrupted disk is an expected failure condition, we should always shutdown if we do not have a working tower.

Reply:
however i haven't really had to debug these scenarios before, if you think the unwind from panic is helpful we can go with that.

Reply:
I think the difference is small, but panic! is probably better style because we use it more:
https://stackoverflow.com/questions/39228685/when-is-stdprocessexit-o-k-to-use